perlapi, perlintern: Show all function signatures

khwilliamson opened the pull request:
Suppose someone wants to use the long name of an element in these pods.  I realized that there was no way to know its signature, or even if such a name exists without also looking through the source code.
     
For example, macros don't have long names, and that an item is is a macro or not was not shown.  Some long names take a pTHX, and some don't.  This also wasn't shown.
     
This commit simply shows all legal signatures, eliminating all ambiguity.  It enables not having to output the warning messages that certain forms aren't available; as that fact is immediately known.

khwilliamson committed:
autodoc: Vertically align long/short usage names

perlapi and perlintern now output both the short and long name
prototypes for every API element that has them.  This commit adds code
so that every element in a group is nicely vertically aligned, so that
any 'Perl_' prefix is outdented, and the basenames are vertically
aligned, along with the arguments.

khwilliamson commented:
There was a bug here in an edge case, so, I fixed it and repushed after rebasing.
The bug was that either the first argument was that you might not want to indent the full 6 character that `pTHX_ ' occupies if the next line is to have a hanging indent less than that.

jkeenan commented:
> There was a bug here in an edge case, so, I fixed it and repushed after rebasing. The bug was that either the first argument was that you might not want to indent the full 6 character that `pTHX_ ' occupies if the next line is to have a hanging indent less than that.

I don't understand the 2nd sentence in the paragraph above.  Among other things, the `either` is not matched with an `or`. Can you clarify? Thanks.

khwilliamson committed:
perlapi, perlintern: Show all function signatures

Suppose someone wants to use the long name of an element in these pods.
I realized that there was no way to know its signature, or even if such
a name exists without also looking through the source code.

For example, macros don't have long names, and that an item is is a
macro or not was not shown.  Some long names take a pTHX, and some
don't.  This also wasn't shown.

This commit simply shows all legal signatures, eliminating all
ambiguity.  It enables not having to output the warning messages that
certain forms aren't available; as that fact is immediately known.

khwilliamson commented:
I edited my text above, but a picture is better.  This only affects a few items.  Before the fix, this one looked like
````
 void       cv_set_call_checker_flags(CV *cv,
                                          Perl_call_checker ckfun,
                                          SV *ckobj, U32 ckflags)
 void  Perl_cv_set_call_checker_flags(pTHX_ CV *cv,
                                          Perl_call_checker ckfun,
                                          SV *ckobj, U32 ckflags)
````
and after the fix
````
 void       cv_set_call_checker_flags(    CV *cv,
                                          Perl_call_checker ckfun,
                                          SV *ckobj, U32 ckflags)
 void  Perl_cv_set_call_checker_flags(pTHX_ CV *cv,
                                          Perl_call_checker ckfun,
                                          SV *ckobj, U32 ckflags)
````
The difference is the `CV *cv` in the very first line isn't vertically aligned with the other arguments
